### ISSUE TITLE
RNGH crashes when enableHermes is disabled on RN 0.72

### ISSUE THREAD

**Issue by alexhernandez** [software-mansion/react-native-gesture-handler#2560]
### Description

Hi All 👋 ,
Big fan of your open source work!

Came here to report an issue w/ RNGH on a fresh React Native 0.72 install. The issue is Android specific and only occurs when using RNGH and `enableHermes` is set to `false`.

I've created a GitHub example link as well:
https://github.com/alexhernandez/rngh-crash-android-example

### Steps to reproduce

1. `npx react-native@latest init rngh-crash-test`
2. Open Android > `gradle.properties` and set `enableHermes` to `false`
3. Add `import 'react-native-gesture-handler'` to the root RN `index.js` file
4. Build the app w/ `yarn android` and notice the red screen error when the app finally builds
5. The crash only occurs on Android when `enableHermes` is set to `false`. Works when `enableHermes` is set to `true`

<img width="520" alt="Screenshot 2023-07-31 at 11 52 16 AM" src="https://github.com/software-mansion/react-native-gesture-handler/assets/2424355/1effb22e-f400-4168-873b-b81d87816360">

### Snack or a link to a repository

https://github.com/alexhernandez/rngh-crash-android-example

### Gesture Handler version

2.12.1

### React Native version

0.72.3

### Platforms

Android, iOS

### JavaScript runtime

JSC

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Android emulator

### Acknowledgements

Yes

**Comment by j-piasecki** [comment]
Hi! Your repro is working fine for me:

https://github.com/software-mansion/react-native-gesture-handler/assets/21055725/df2201be-ee88-479b-a5cc-56ceb8efe66f

The result is the same when running on API 33 emulator.

**Comment by alexhernandez** [comment]
> Hi! Your repro is working fine for me:
> 
> 
> 
> https://github.com/software-mansion/react-native-gesture-handler/assets/21055725/df2201be-ee88-479b-a5cc-56ceb8efe66f
> 
> 
> 
> The result is the same when running on API 33 emulator. 

For your bare install enableHermes is false correct?

Could you share what you get when you run:

```
npx react-native info
```

**Comment by AndriyZahura** [comment]
Not sure if it's the same error, but I had the same message. It was coming from the latest react-native-reanimated version, and it was because of use of `&&=` operator in the package. Take a look at the reanimated issue
https://github.com/software-mansion/react-native-reanimated/issues/4825

**Comment by j-piasecki** [comment]
> Could you share your Android Studio version and what you get when you run:

Android Studio Flamingo | 2022.2.1 Patch 2

```
System:
  OS: macOS 12.6
  CPU: (8) arm64 Apple M1 Pro
  Memory: 303.31 MB / 16.00 GB
  Shell:
    version: 5.8.1
    path: /bin/zsh
Binaries:
  Node:
    version: 16.15.1
    path: ~/.nvm/versions/node/v16.15.1/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 8.11.0
    path: ~/.nvm/versions/node/v16.15.1/bin/npm
  Watchman:
    version: 2023.01.16.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods: Not Found
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.2
      - iOS 16.2
      - macOS 13.1
      - tvOS 16.1
      - watchOS 9.1
  Android SDK:
    API Levels:
      - "24"
      - "26"
      - "28"
      - "29"
      - "30"
      - "31"
      - "32"
      - "33"
    Build Tools:
      - 26.0.3
      - 28.0.3
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 32.0.0
      - 32.1.0
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-28 | Google ARM64-V8a Play ARM 64 v8a
      - android-33 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.10121639
  Xcode:
    version: 14.2/14C18
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.14
    path: /usr/bin/javac
  Ruby:
    version: 2.7.5
    path: /Users/jakubpiasecki/.rvm/rubies/ruby-2.7.5/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.3
    wanted: 0.72.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: false
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: Not found
```

**Comment by vasylnahuliak** [comment]
Fixed in react-native-reanimated@3.4.2


```
npm install react-native-reanimated@3.4.2
```

https://github.com/software-mansion/react-native-reanimated/releases/tag/3.4.2

**Comment by alexhernandez** [comment]
Thanks @vasylnahuliak updating my version of RNGH and reanimated fixed my issue!

RNGH: v2.12.1
Reanimated: 3.4.2
